mutable_tree.go

```diff
@@ -85,21 +85,24 @@ func (tree *MutableTree) Set(key, value []byte) bool {
 	return updated
 }
 
-func (tree *MutableTree) set(key []byte, value []byte) (orphaned []*Node, updated bool) {
+func (tree *MutableTree) set(key []byte, value []byte) ([]*Node, bool) {
 	if value == nil {
 		panic(fmt.Sprintf("Attempt to store nil value at key '%s'", key))
 	}
+
+	updated := false
 	if tree.ImmutableTree.root == nil {
 		tree.ImmutableTree.root = NewNode(key, value, tree.version+1)
-		return nil, false
+		return nil, updated
 	}
-	tree.ImmutableTree.root, updated, orphaned = tree.recursiveSet(tree.ImmutableTree.root, key, value)
 
+	orphaned := make([]*Node, 0, tree.Height()*2)
```

Comment:
Why `tree.Height() * 2`? Shouldn't it be `tree.Size()` or something?

Reply:
`recursiveSet` will orphan every node that it recurses upon. Thus, this will orphan `tree.Height` nodes.

I'm guessing the `* 2` is from the balance call at the end? However, if you look at balance implementation; it orphans at most 3 nodes.

Thus, I believe this line should be `tree.Height + 3`. Someone should definitely check my math on this.

Regardless, definitely needs a docstring here explaining how we get this number.

Reply:
right, 
`recursiveSet` will orphan at most `tree.Height()` nodes, `balance` will produce at most 3 nodes. so `tree.Height() + 3` is enough.

Reply:
I think it may make sense to make this a helper function that returns the orphan slice for you. Makes the code slightly more DRY making it easier to update if the calculation ever changes, but also can have documentation on that helper function.